tests/test_serialization.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017-2019- Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import yaml
+import datetime
+
+
+def test_dataset_serialization(client, dataset, data_file):
+    """Test Dataset serialization."""
+    def load_dataset(name):
+        with open(client.dataset_path(name)) as f:
+            return yaml.safe_load(f)
+
+    d_dict = load_dataset('dataset')
+
+    expected_fields = [
+        '_id', '_label', '_project', 'created', 'creator', 'date_published',
+        'description', 'files', 'identifier', 'in_language', 'keywords',
+        'license', 'name', 'path', 'url', 'version'
+    ]
+    for field in expected_fields:
+        assert field in d_dict
+
+    assert not d_dict['files']
+    client.add_data_to_dataset(dataset, [str(data_file)])
+    dataset.to_yaml()
+    d_dict = load_dataset('dataset')
+    assert d_dict['files']
+
+
+def test_dataset_deserialization(client, dataset):
+    """Test Dataset deserialization."""
+    dataset_ = client.get_dataset(client.dataset_path('dataset'))
```

Comment:
Can we load just like we did in serialization test? We might need to attach additional validation checks on `get_dataset` method in the future.

Reply:
done